DiffusionPreprocessing/scripts/eddy_postproc.sh

```diff
@@ -31,70 +33,80 @@ qc_command+=(-v)
 "${qc_command[@]}"
 
 #Prepare for next eddy Release
-#if [ ! -e ${eddydir}/${EddyJacFlag} ]; then 
+#if [ ! -e ${eddydir}/${EddyJacFlag} ]; then
 #    echo "LSR resampling has been used. Eddy Output has already been combined."
 #    cp ${eddydir}/Pos.bval ${datadir}/bvals
 #    cp ${eddydir}/Pos.bvec ${datadir}/bvecs
 #    $FSLDIR/bin/imcp ${eddydir}/eddy_unwarped_images ${datadir}/data
 #else
+if [ ${SelectBestB0} -eq 1 ]; then
+	cut -d' ' -f2- ${eddydir}/Pos_Neg.bvals >${datadir}/bvals       # removes first value from bvals
+	cut -d' ' -f2- ${eddydir}/Pos_Neg.bvecs >${datadir}/bvecs_noRot # removes first value from bvecs
+fi
 if [ ${CombineDataFlag} -eq 2 ]; then
-	${FSLDIR}/bin/imcp  ${eddydir}/eddy_unwarped_images ${datadir}/data
-	cp ${eddydir}/Pos_Neg.bvals ${datadir}/bvals
-	cp ${datadir}/bvecs ${datadir}/bvecs_noRot
-	cp ${eddydir}/eddy_unwarped_images.eddy_rotated_bvecs ${datadir}/bvecs
+	# remove first volume as this is the reference b0, which was added to the dataset before running eddy
+	if [ ${SelectBestB0} -eq 1 ]; then
+		${FSLDIR}/bin/fslroi ${eddydir}/eddy_unwarped_images ${datadir}/data 1 -1
+		cut -d' ' -f2- ${eddydir}/eddy_unwarped_images.eddy_rotated_bvecs ${datadir}/bvecs # removes first value from bvecs
+	else
+		${FSLDIR}/bin/imcp ${eddydir}/eddy_unwarped_images ${datadir}/data
+		cp ${eddydir}/Pos_Neg.bvals ${datadir}/bvals
+		cp ${datadir}/bvecs ${datadir}/bvecs_noRot
+		cp ${eddydir}/eddy_unwarped_images.eddy_rotated_bvecs ${datadir}/bvecs
+	fi
 else
 	echo "JAC resampling has been used. Eddy Output is now combined."
-	PosVols=`wc ${eddydir}/Pos.bval | awk {'print $2'}`
-	NegVols=`wc ${eddydir}/Neg.bval | awk {'print $2'}`    #Split Pos and Neg Volumes
-	${FSLDIR}/bin/fslroi ${eddydir}/eddy_unwarped_images ${eddydir}/eddy_unwarped_Pos 0 ${PosVols}
-	${FSLDIR}/bin/fslroi ${eddydir}/eddy_unwarped_images ${eddydir}/eddy_unwarped_Neg ${PosVols} ${NegVols}
+	PosVols=$(wc ${eddydir}/Pos.bval | awk {'print $2'})
+	NegVols=$(wc ${eddydir}/Neg.bval | awk {'print $2'})                                                         #Split Pos and Neg Volumes
```

Comment:
Seems to have been a lot of space added between the end of the command and the comment (possibly by pycharm)?